src/main/java/seedu/uninurse/model/person/PatientMatchPredicate.java

```diff
@@ -0,0 +1,90 @@
+    private final NameContainsKeywordsPredicate nameContainsKeywordsPredicate;
+    private final PhoneContainsKeywordsPredicate phoneContainsKeywordsPredicate;
+    private final EmailContainsKeywordsPredicate emailContainsKeywordsPredicate;
+    private final AddressContainsKeywordsPredicate addressContainsKeywordsPredicate;
+    private final TagContainsKeywordsPredicate tagContainsKeywordsPredicate;
+    private final ConditionContainsKeywordsPredicate conditionContainsKeywordsPredicate;
+    private final TaskContainsKeywordsPredicate taskContainsKeywordsPredicate;
+
+    /**
+     * Constructs a {@code PatientMatchPredicate}
+     * which tests {@code Patient} with given keywords.
+     *
+     * @param keywords
+     */
+    public PatientMatchPredicate(List<String> keywords) {
+        this(keywords, List.of(""), List.of(""), List.of(""),
+                List.of(""), List.of(""), List.of(""), List.of(""));
+    }
+
+    /**
+     * Constructs a {@code PatientMatchPredicate}
+     * which tests {@code Patient} with all given descriptors.
+     *
+     * @param keywords
+     * @param nameList
+     * @param phoneList
+     * @param emailList
+     * @param addressList
+     * @param tagList
+     * @param conditionList
+     * @param taskList
+     */
+    public PatientMatchPredicate(List<String> keywords, List<String> nameList, List<String> phoneList,
+                                 List<String> emailList, List<String> addressList, List<String> tagList,
+                                 List<String> conditionList, List<String> taskList) {
+        this.patientContainsKeywordsPredicate = new PatientContainsKeywordsPredicate(keywords);
+        this.nameContainsKeywordsPredicate = new NameContainsKeywordsPredicate(nameList);
+        this.phoneContainsKeywordsPredicate = new PhoneContainsKeywordsPredicate(phoneList);
+        this.emailContainsKeywordsPredicate = new EmailContainsKeywordsPredicate(emailList);
+        this.addressContainsKeywordsPredicate = new AddressContainsKeywordsPredicate(addressList);
+        this.tagContainsKeywordsPredicate = new TagContainsKeywordsPredicate(tagList);
+        this.conditionContainsKeywordsPredicate = new ConditionContainsKeywordsPredicate(conditionList);
+        this.taskContainsKeywordsPredicate = new TaskContainsKeywordsPredicate(taskList);
+    }
+
+    @Override
+    public boolean test(Patient person) {
+        return patientContainsKeywordsPredicate.test(person)
+                && nameContainsKeywordsPredicate.test(person)
+                && phoneContainsKeywordsPredicate.test(person)
+                && emailContainsKeywordsPredicate.test(person)
+                && addressContainsKeywordsPredicate.test(person)
+                && tagContainsKeywordsPredicate.test(person)
+                && conditionContainsKeywordsPredicate.test(person)
+                && taskContainsKeywordsPredicate.test(person);
+    }
```

Comment:
If you use a `List<Predicate<Patient>>`  you can probably just streamify it here.

Reply:
Done